### ISSUE TITLE
`git reflow deliver` no longer squashing!

### ISSUE THREAD

**Issue by jmduke** [reenhanced/gitreflow#212]
Hi! I recently bumped up to `0.8.6` and now when I run `git reflow deliver` on a valid branch reflow errors out with the following message:

```
error: PUT https://api.github.com/repos/TenantBase/tenantbase/pulls/1003/merge?access_token=77a741a8a73d4cca2db6baff5df5e5ddad558ca9: 405 - Merge commits are not allowed on this repository.
See: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button
```

Naturally, we can enable straight merges for the repo but then the behavior from before (squashing all commits on the branch into master) is no longer present.

It looks like there was some recent work on merge methods, and that may have something to do with it?

https://github.com/reenhanced/gitreflow/commit/3a766a0e9617b1ecd7a711a988c4a138aa4571d5
https://github.com/reenhanced/gitreflow/issues/201
https://github.com/reenhanced/gitreflow/pull/211/files

Thanks so much for all your hard work on this project!

**Comment by Kobold** [comment]
To add some additional information, this looks to be a Github API issue.

I've been using reflow 0.8.3, and in the last few days `git reflow deliver` has started failing with that `405 - Merge commits are not allowed on this repository.` above. Again, on version 0.8.3.

Updating to reflow 0.8.6, `git reflow deliver` still fails with `405 - Merge commits are not allowed on this repository.`

It appears to be this github API update as of April 7th: https://developer.github.com/changes/2017-04-07-end-merge-methods-api-preview/

> Note: During the preview period, we initially added a squash property in the payload to the pull request merge endpoint. This property was superseded by a merge_method property to support both "squash and merge" as well as "rebase and merge" operations. With the end of the preview period, support for the squash property was removed in favor of the merge_method property.

It appears PRs #205 and #211 together provide a fix for this API change. Can you folks consider cutting a new release with the fix? Thanks!

**Comment by Kobold** [comment]
And to aid replication efforts on your end, here are the settings for our repository where the error is occurring. Note that we've specifically disabled merge commits, and this has worked well with reflow historically.

![options](https://cloud.githubusercontent.com/assets/36694/24984000/992d6504-1f9f-11e7-9fca-942ced6077d3.png)

**Comment by nhance** [comment]
This has been fixed in `HEAD` but has not yet been released. I can update to `0.8.7` today which should solve this

**Comment by codenamev** [comment]
This is resolved in `0.8.8`.

**Comment by Kobold** [comment]
Thanks again @codenamev !
